### ISSUE TITLE
Error with fine-tune

### ISSUE THREAD

**Issue by h4rk8s** [johannakarras/DreamPose#33]
After running latest script from `Readme.md` 👇🏻

```bash
accelerate launch finetune-unet.py --pretrained_model_name_or_path="CompVis/stable-diffusion-v1-4" --instance_data_dir=demo/sample/train --output_dir=demo/custom-chkpts --resolution=512 --train_batch_size=1 --gradient_accumulation_steps=1 --learning_rate=1e-5 --num_train_epochs=500 --dropout_rate=0.0 --custom_chkpt=checkpoints/unet_epoch_20.pth --revision "ebb811dd71cdc38a204ecbdd6ac5d580f529fd8c"


accelerate launch --num_processes=1 finetune-vae.py --pretrained_model_name_or_path="CompVis/stable-diffusion-v1-4"  --instance_data_dir=demo/sample/train --output_dir=demo/custom-chkpts --resolution=512  --train_batch_size=4 --gradient_accumulation_steps=4 --learning_rate=5e-5 --num_train_epochs=1500 --run_name finetuning/ubc-vae --revision "ebb811dd71cdc38a204ecbdd6ac5d580f529fd8c"


python test.py --epoch 499 --folder demo/custom-chkpts --pose_folder demo/sample/poses  --key_frame_path demo/sample/key_frame.png --s1 8 --s2 3 --n_steps 100 --output_dir results --custom_vae demo/custom-chkpts/vae_1499.pth
```
`results/` OUTPUT:
![pred_#50](https://user-images.githubusercontent.com/63350517/236418239-9507f3b6-a133-4ead-b9c8-b03a4abfc329.png)

I test the new scripts, but the problem still exists unfortunately.

this is my env build script :
```bash
#!/bin/bash
set -e

CONDA_P=${1-"/workspace/miniconda3"}
CONDA_BIN=$CONDA_P/bin
ENV=ldm
if [ ! -d $CONDA_P/envs/$ENV ]; then
    $CONDA_BIN/conda create --name $ENV python=3.10 -y
fi
ENV_BIN=$CONDA_P/envs/$ENV/bin

$ENV_BIN/pip install --upgrade diffusers[torch]
$ENV_BIN/pip install torchvision --index-url https://download.pytorch.org/whl/cu117

echo misc
conda install -n $ENV click scipy tqdm ninja matplotlib imageio pandas -y
for pkg in \
    imageio-ffmpeg opencv-python-headless omegaconf einops transformers chardet tensorboard
do
    $ENV_BIN/pip install $pkg
done

```

this is env export:

```yaml
name: ldm
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - appdirs=1.4.4=pyhd3eb1b0_0
  - blas=1.0=mkl
  - bottleneck=1.3.5=py310ha9d4c09_0
  - brotli=1.0.9=h5eee18b_7
  - brotli-bin=1.0.9=h5eee18b_7
  - brotlipy=0.7.0=py310h7f8727e_1002
  - bzip2=1.0.8=h7b6447c_0
  - ca-certificates=2023.01.10=h06a4308_0
  - certifi=2022.12.7=py310h06a4308_0
  - cffi=1.15.1=py310h5eee18b_3
  - click=8.0.4=py310h06a4308_0
  - contourpy=1.0.5=py310hdb19cb5_0
  - cryptography=39.0.1=py310h9ce1e76_0
  - cycler=0.11.0=pyhd3eb1b0_0
  - dbus=1.13.18=hb2f20db_0
  - expat=2.4.9=h6a678d5_0
  - fontconfig=2.14.1=h4c34cd2_2
  - fonttools=4.25.0=pyhd3eb1b0_0
  - freetype=2.12.1=h4a9f257_0
  - giflib=5.2.1=h5eee18b_3
  - glib=2.69.1=he621ea3_2
  - gst-plugins-base=1.14.1=h6a678d5_1
  - gstreamer=1.14.1=h5eee18b_1
  - icu=58.2=he6710b0_3
  - idna=3.4=py310h06a4308_0
  - imageio=2.26.0=py310h06a4308_0
  - intel-openmp=2021.4.0=h06a4308_3561
  - jpeg=9e=h5eee18b_1
  - kiwisolver=1.4.4=py310h6a678d5_0
  - krb5=1.19.4=h568e23c_0
  - lcms2=2.12=h3be6417_0
  - ld_impl_linux-64=2.38=h1181459_1
  - lerc=3.0=h295c915_0
  - libbrotlicommon=1.0.9=h5eee18b_7
  - libbrotlidec=1.0.9=h5eee18b_7
  - libbrotlienc=1.0.9=h5eee18b_7
  - libclang=14.0.6=default_hc6dbbc7_1
  - libclang13=14.0.6=default_he11475f_1
  - libdeflate=1.17=h5eee18b_0
  - libedit=3.1.20221030=h5eee18b_0
  - libevent=2.1.12=h8f2d780_0
  - libffi=3.4.2=h6a678d5_6
  - libgcc-ng=11.2.0=h1234567_1
  - libgfortran-ng=11.2.0=h00389a5_1
  - libgfortran5=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libllvm14=14.0.6=hdb19cb5_2
  - libpng=1.6.39=h5eee18b_0
  - libpq=12.9=h16c4e8d_3
  - libstdcxx-ng=11.2.0=h1234567_1
  - libtiff=4.5.0=h6a678d5_2
  - libuuid=1.41.5=h5eee18b_0
  - libwebp=1.2.4=h11a3e52_1
  - libwebp-base=1.2.4=h5eee18b_1
  - libxcb=1.15=h7f8727e_0
  - libxkbcommon=1.0.1=h5eee18b_1
  - libxml2=2.10.3=hcbfbd50_0
  - libxslt=1.1.37=h2085143_0
  - lz4-c=1.9.4=h6a678d5_0
  - matplotlib=3.7.1=py310h06a4308_1
  - matplotlib-base=3.7.1=py310h1128e8f_1
  - mkl=2021.4.0=h06a4308_640
  - mkl-service=2.4.0=py310h7f8727e_0
  - mkl_fft=1.3.1=py310hd6ae3a3_0
  - mkl_random=1.2.2=py310h00e6091_0
  - munkres=1.1.4=py_0
  - ncurses=6.4=h6a678d5_0
  - ninja=1.10.2=h06a4308_5
  - ninja-base=1.10.2=hd09550d_5
  - nspr=4.33=h295c915_0
  - nss=3.74=h0370c37_0
  - numexpr=2.8.4=py310h8879344_0
  - numpy=1.24.3=py310hd5efca6_0
  - numpy-base=1.24.3=py310h8e6c178_0
  - openssl=1.1.1t=h7f8727e_0
  - pandas=1.5.3=py310h1128e8f_0
  - pcre=8.45=h295c915_0
  - pip=23.0.1=py310h06a4308_0
  - ply=3.11=py310h06a4308_0
  - pooch=1.4.0=pyhd3eb1b0_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyopenssl=23.0.0=py310h06a4308_0
  - pyparsing=3.0.9=py310h06a4308_0
  - pyqt=5.15.7=py310h6a678d5_1
  - pysocks=1.7.1=py310h06a4308_0
  - python=3.10.11=h7a1cb2a_2
  - python-dateutil=2.8.2=pyhd3eb1b0_0
  - pytz=2022.7=py310h06a4308_0
  - qt-main=5.15.2=h8373d8f_8
  - qt-webengine=5.15.9=hbbf29b9_6
  - qtwebkit=5.212=h3fafdc1_5
  - readline=8.2=h5eee18b_0
  - scipy=1.10.1=py310hd5efca6_0
  - setuptools=66.0.0=py310h06a4308_0
  - sip=6.6.2=py310h6a678d5_0
  - six=1.16.0=pyhd3eb1b0_1
  - sqlite=3.41.2=h5eee18b_0
  - tk=8.6.12=h1ccaba5_0
  - toml=0.10.2=pyhd3eb1b0_0
  - tornado=6.2=py310h5eee18b_0
  - tqdm=4.65.0=py310h2f386ee_0
  - tzdata=2023c=h04d1e81_0
  - wheel=0.38.4=py310h06a4308_0
  - xz=5.4.2=h5eee18b_0
  - zlib=1.2.13=h5eee18b_0
  - zstd=1.5.5=hc292b87_0
  - pip:
      - absl-py==1.4.0
      - accelerate==0.18.0
      - antlr4-python3-runtime==4.9.3
      - cachetools==5.3.0
      - chardet==5.1.0
      - charset-normalizer==3.1.0
      - cmake==3.26.3
      - diffusers==0.16.1
      - einops==0.6.1
      - filelock==3.12.0
      - fsspec==2023.4.0
      - google-auth==2.17.3
      - google-auth-oauthlib==1.0.0
      - grpcio==1.54.0
      - huggingface-hub==0.14.1
      - imageio-ffmpeg==0.4.8
      - importlib-metadata==6.6.0
      - jinja2==3.1.2
      - lit==16.0.2
      - markdown==3.4.3
      - markupsafe==2.1.2
      - mpmath==1.3.0
      - networkx==3.1
      - nvidia-cublas-cu11==11.10.3.66
      - nvidia-cuda-cupti-cu11==11.7.101
      - nvidia-cuda-nvrtc-cu11==11.7.99
      - nvidia-cuda-runtime-cu11==11.7.99
      - nvidia-cudnn-cu11==8.5.0.96
      - nvidia-cufft-cu11==10.9.0.58
      - nvidia-curand-cu11==10.2.10.91
      - nvidia-cusolver-cu11==11.4.0.1
      - nvidia-cusparse-cu11==11.7.4.91
      - nvidia-nccl-cu11==2.14.3
      - nvidia-nvtx-cu11==11.7.91
      - oauthlib==3.2.2
      - omegaconf==2.3.0
      - opencv-python-headless==4.7.0.72
      - packaging==23.1
      - pillow==9.5.0
      - protobuf==4.22.4
      - psutil==5.9.5
      - pyasn1==0.5.0
      - pyasn1-modules==0.3.0
      - pyqt5-sip==12.11.0
      - pyyaml==6.0
      - regex==2023.5.5
      - requests==2.30.0
      - requests-oauthlib==1.3.1
      - rsa==4.9
      - sympy==1.11.1
      - tensorboard==2.13.0
      - tensorboard-data-server==0.7.0
      - tokenizers==0.13.3
      - torch==2.0.0
      - torchvision==0.15.1+cu117
      - transformers==4.28.1
      - triton==2.0.0
      - typing-extensions==4.5.0
      - urllib3==2.0.2
      - werkzeug==2.3.3
      - zipp==3.15.0
prefix: /workspace/miniconda3/envs/ldm
```

May be it's env issule ? can you sharing you'are env ? @johannakarras

**Comment by johannakarras** [comment]
Thank you for your feedback! I believe this is an issue with the normalization of the output image. I have fixed and updated the code in test.py. Please try again with that script.
